src/Cli/dotnet/SlnFileFactory.cs

```diff
@@ -1,33 +1,48 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using Microsoft.DotNet.Cli.Sln.Internal;
 using Microsoft.DotNet.Cli.Utils;
+using Microsoft.VisualStudio.SolutionPersistence;
+using Microsoft.VisualStudio.SolutionPersistence.Model;
+using Microsoft.VisualStudio.SolutionPersistence.Serializer;
 
 namespace Microsoft.DotNet.Tools.Common
 {
     public static class SlnFileFactory
     {
-        public static SlnFile CreateFromFileOrDirectory(string fileOrDirectory)
+        public static string[] ListSolutionFilesInDirectory(string directory, bool includeSolutionFilterFiles = false, bool includeSolutionXmlFiles = true)
+        {
+            return [
+                ..Directory.GetFiles(directory, "*.sln", SearchOption.TopDirectoryOnly),
+                ..(includeSolutionXmlFiles ? Directory.GetFiles(directory, "*.slnx", SearchOption.TopDirectoryOnly) : Array.Empty<string>()),
+                ..(includeSolutionFilterFiles ? Directory.GetFiles(directory, "*.slnf", SearchOption.TopDirectoryOnly) : Array.Empty<string>())
```

Comment:
nit: I think `[]` would work here instead of `Array.Empty<string>()`. It should be aware of the LHS (left-hand side) type.